solar-system/src/org/gearvrf/solarsystem/LightMapManager.java

```diff
@@ -0,0 +1,181 @@
+            + "}";
+
+        String fragment_shader = "precision mediump float;\n"
+            + "varying vec2  coord;\n"
+            + "uniform sampler2D main_texture;\n"
+            + "uniform sampler2D lightmap_texture;\n"
+            + "uniform vec2  lightmap_offset;\n"
+            + "uniform vec2  lightmap_scale;\n"
+            + "void main() {\n"
+            + " vec4 color;\n"
+            + " vec4 lightmap_color;\n"
+            + " vec2 lightmap_coord = (coord * lightmap_scale) + lightmap_offset;\n"
+            // Beast exports the texture with vertical flip
+            + " lightmap_color = texture2D(lightmap_texture, vec2(lightmap_coord.x, 1.0 - lightmap_coord.y));\n"
+            + " color = texture2D(main_texture, coord);\n"
+            + " gl_FragColor = color * lightmap_color;\n"
+            + "}";
+
+        public static final String MAIN_TEXTURE = "main_texture";
+        public static final String LIGHTMAP_TEXTURE = "lightmap_texture";
+        public static final String LIGHTMAP_OFFSET = "lightmap_offset";
+        public static final String LIGHTMAP_SCALE = "lightmap_scale";
+        public static final String LIGHTMAP_INTENSITY = "lightmap_intensity";
+
+        private GVRCustomMaterialShaderId mShaderId;
+        private GVRMaterialMap mCustomShader = null;
+
+        public LightMapShader(GVRContext gvrContext) {
+            final GVRMaterialShaderManager shaderManager = gvrContext
+                .getMaterialShaderManager();
+
+            mShaderId = shaderManager.addShader(vertex_shader, fragment_shader);
+
+            mCustomShader = shaderManager.getShaderMap(mShaderId);
+            mCustomShader.addTextureKey(MAIN_TEXTURE, MAIN_TEXTURE);
+            mCustomShader.addTextureKey(LIGHTMAP_TEXTURE, LIGHTMAP_TEXTURE);
+            mCustomShader.addUniformVec2Key(LIGHTMAP_OFFSET, LIGHTMAP_OFFSET);
+            mCustomShader.addUniformVec2Key(LIGHTMAP_SCALE, LIGHTMAP_SCALE);
+        }
+
+        public GVRCustomMaterialShaderId getShaderId() {
+            return mShaderId;
+        }
+    }
+
+    protected class LightMapInfo {
+
+        public class LightMapInfoObject {
+            public String name;
+            public float offsetX;
+            public float offsetY;
+            public float scaleX;
+            public float scaleY;
+        }
+
+        private List<LightMapInfoObject> mMapInfoList;
+
+        public LightMapInfo(GVRContext gvrContext, int lighMapInfoId) {
+            mMapInfoList = new ArrayList<LightMapInfoObject>();
+
+            loadLightMapInfo(gvrContext, lighMapInfoId);
+        }
+
+        private void loadLightMapInfo(GVRContext gvrContext, int lighMapInfoId) {
+            try {
+                InputStream is = gvrContext.getActivity().getResources().openRawResource(lighMapInfoId);
+                int size = is.available();
+                byte[] buffer = new byte[size];
+
+                is.read(buffer);
+
+                is.close();
```

Comment:
Please move is.close() to finally block (also catch IOException for each close() call).